app/src/main/java/ingsw/group1/findmyphone/ResponseManager.java

```diff
@@ -0,0 +1,182 @@
+import androidx.lifecycle.MutableLiveData;
+
+import com.eis.smslibrary.SMSManager;
+import com.eis.smslibrary.SMSMessage;
+import com.eis.smslibrary.SMSPeer;
+
+import ingsw.group1.findmyphone.activity.NavHolderActivity;
+import ingsw.group1.findmyphone.contacts.SMSContactManager;
+import ingsw.group1.findmyphone.event.EventType;
+import ingsw.group1.findmyphone.event.SMSLogDatabase;
+import ingsw.group1.findmyphone.event.SMSLogEvent;
+import ingsw.group1.findmyphone.location.GeoPosition;
+import ingsw.group1.findmyphone.location.LocationManager;
+import ingsw.group1.findmyphone.location.LocationResponseCommand;
+import ingsw.group1.findmyphone.log.LogManager;
+import ingsw.group1.findmyphone.log.items.LogItem;
+import ingsw.group1.findmyphone.log.items.LogItemFormatter;
+
+/**
+ * Class meant to manage responses to the various incoming messages. This includes starting the
+ * appropriate activities/services if necessary.
+ *
+ * @author Riccardo De Zen.
+ */
+public class ResponseManager {
+
+    private static ResponseManager instance;
+
+    private LocationManager locationManager = new LocationManager();
+    private MessageParser messageParser = new MessageParser();
+    private SMSContactManager contacts;
+    private SMSLogDatabase logDatabase;
+    private LogItemFormatter logItemFormatter;
+
+    /**
+     * Default constructor, private in order to be hidden from the user.
+     */
+    private ResponseManager(Context context) {
+        logDatabase = SMSLogDatabase.getInstance(context, LogManager.DEFAULT_LOG_DATABASE);
+        contacts = SMSContactManager.getInstance(context);
+        logItemFormatter = new LogItemFormatter(context);
+    }
+
+    /**
+     * Method to return the only available instance of the class.
+     *
+     * @param context The calling Context.
+     * @return The available instance of this class.
+     */
+    public static ResponseManager getInstance(Context context) {
+        if (instance != null) return instance;
+        instance = new ResponseManager(context.getApplicationContext());
+        return instance;
+    }
+
+    /**
+     * The message is assumed to have already been decrypted.
+     * Performs the appropriate action based on what the type of the message is:
+     * - {@link MessageType#RING_REQUEST} The foreground Service is started.
+     * - {@link MessageType#RING_RESPONSE} The Ring Event is recorded in the Log Database.
+     * - {@link MessageType#LOCATION_REQUEST} The asynchronous location retrieval is started and
+     * will send back the result when available.
+     * - {@link MessageType#LOCATION_RESPONSE} The Location Event is recorded in the Log Database.
+     *
+     * @param callingContext The calling context, used if starting a Service or such.
+     * @param message        The message to handle.
+     */
+    public void performActionBasedOnMessage(Context callingContext, @NonNull SMSMessage message) {
+        switch (messageParser.getMessageType(message)) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Switch statements should have a default label](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5114829)